### ISSUE TITLE
Enhance diff_check with more unicode et al

### ISSUE THREAD

**Issue by calebcartwright** [rust-lang/rustfmt#5884]
Follow up action item from a topic I raised in https://github.com/rust-lang/rustfmt/pull/5864#issuecomment-1676425492 relative to adding some additional test repos to improve our coverage of string formatting with various characters and encodings.

One suggested repo to include is https://github.com/unicode-org/icu4x as well as some from https://github.com/unicode-rs

I opted to simply mention icu4x as part of this instead of directly adding it because I noticed that repo is using `cargo make` and I didn't have bandwidth to determine if that'd cause any issues/require any additional changes to our script.

New repos can be wired up by adding a call site with respective parameters here:

https://github.com/rust-lang/rustfmt/blob/b069aac44ddfdb70d55d9ae40695be44515e5bb0/ci/check_diff.sh#L170-L192

**Comment by ytmimi** [comment]
@calebcartwright I don't think the use of `cargo make` on any project we test would be an issue. The diff check job just runs rustfmt againsg all `.rs` files it finds in the repo. 

https://github.com/rust-lang/rustfmt/blob/f89cd3c1f3feffaa960e13c7bec07a6163d18fd3/ci/check_diff.sh#L59-L62

**Comment by calebcartwright** [comment]
Forgot we were running `rustfmt` directly. The only potential downside to this would be our ability to pull in any target repositories that needed 2018+ edition & didn't have `edition` defined in a `rustfmt.toml` file.

Orthogonal to this particular issue, but at some point would like to think/talk through whether that has any implications once `style_edition` enters the ecosystem

**Comment by ytmimi** [comment]
@calebcartwright good point. I guess if we landed support for `cargo fmt` to be passed individual files then that would mostly solve the `edition` issue. I'm nearly 100% sure `style_edition` is going to be implemented in such a way that we'll derive it from the `--edition` if you don't explicitly configure a different style edition.

**Comment by rufevean** [comment]
@ytmimi , is this issue still open to be worked on as the old assignee doesnt seem like he is working on this anymore?

**Comment by ytmimi** [comment]
@rufevean yeah this one is still open

**Comment by rufevean** [comment]
got it, thank you

**Comment by rufevean** [comment]
i have added the follow repos into ` ./check_diff.sh` file and i executed it . i am receiving an error stating 
`
./check_diff.sh: line 148: /tmp/rustfmt-IVGzB32u/feature_rustfmt: No such file or directory .`
 is this part of the issue or am i missing something? 

```   
    check_repo "https://github.com/unicode-org/icu4x.git" icu4x
    check_repo "https://github.com/unicode-rs/unicode-segmentation.git" unicode-segmentation
    check_repo "https://github.com/unicode-rs/unicode-normalization.git" unicode-normalization
```

**Comment by ytmimi** [comment]
it's unclear what the issue is here. Are you able to run the script on your machine without adding those new repos?

**Comment by rufevean** [comment]
its working perfectly fine when i removed the repos

**Comment by ytmimi** [comment]
[Line 148 is where we try to get the version value for the feature binary](https://github.com/rust-lang/rustfmt/blob/6157568e1cec55ce6df09342ba0ae9686dec99c9/ci/check_diff.sh#L148). Not sure why adding those lines would cause an issue there. The error message suggests that we're failing to build the binary or we're failing to copy it to the temporary directory. Let me know if you're able to investigate further.

**Comment by rufevean** [comment]
> i have added the follow repos into ` ./check_diff.sh` file and i executed it . i am receiving an error stating ` ./check_diff.sh: line 148: /tmp/rustfmt-IVGzB32u/feature_rustfmt: No such file or directory .` is this part of the issue or am i missing something?
> 
> ```
>     check_repo "https://github.com/unicode-org/icu4x.git" icu4x
>     check_repo "https://github.com/unicode-rs/unicode-segmentation.git" unicode-segmentation
>     check_repo "https://github.com/unicode-rs/unicode-normalization.git" unicode-normalization
> ```

okay, i have tested it now, i was wrong. i wasnt specifying the branch correctly last time. it running well this time with all these three repos.
